### PR TITLE
Remove immutable from status and auth

### PR DESCRIPTION
**Pull request by smashercosmo**
**Summary**

First step on closing #3853

**Test plan**

Run existing tests

### CODE REVIEW

File: packages/netlify-cms-core/src/reducers/auth.ts [hunk 1]
```diff
@@ -8,35 +8,37 @@ import {
...
-export const defaultState = fromJS({
+export const defaultState: Auth = {
   isFetching: false,
   user: undefined,
   error: undefined,
-}) as Auth;
+};
 
-const auth = (state = defaultState, action: AuthAction) => {
+const auth = produce((draft: Auth, action: AuthAction) => {
```
Comment (smashercosmo) [inline packages/netlify-cms-core/src/reducers/auth.ts:24]: Hm... maybe it's better to use 'state' instead of 'draft'

**Comment by smashercosmo** [comment]
Hm... seems like something's not ok

**Comment by smashercosmo** [comment]
@erezrokah any idea why e2e tests are failing?

**Comment by smashercosmo** [comment]
Alright, found the reason

**Review by erezrokah — APPROVED** [review]
Great work @smashercosmo!

**Pull request merged**